src/main/java/io/appium/java_client/pagefactory/DefaultElementByBuilder.java

```diff
@@ -119,105 +112,81 @@ private static By buildMobileBy(LocatorGroupStrategy locatorGroupStrategy, Annot
         return defaultBy;
     }
 
-    @Override protected By buildMobileNativeBy() {
+    private By[] getBys(Class<? extends Annotation> singleLocator, Class<? extends Annotation> chainedLocator,
+                        Class<? extends Annotation> allLocator) {
+        AnnotationComparator comparator = new AnnotationComparator();
         AnnotatedElement annotatedElement = annotatedElementContainer.getAnnotated();
-        HowToUseLocators howToUseLocators = annotatedElement.getAnnotation(HowToUseLocators.class);
 
-        if (isSelendroidAutomation()) {
-            SelendroidFindBy[] selendroidFindByArray =
-                annotatedElement.getAnnotationsByType(SelendroidFindBy.class);
-            //should be kept for some time
-            SelendroidFindBys selendroidFindBys =
-                annotatedElement.getAnnotation(SelendroidFindBys.class);
-            SelendroidFindAll selendroidFindByAll =
-                annotatedElement.getAnnotation(SelendroidFindAll.class);
-
-            if (selendroidFindByArray != null && selendroidFindByArray.length == 1) {
-                return createBy(new Annotation[] {selendroidFindByArray[0]}, HowToUseSelectors.USE_ONE);
-            }
+        List<Annotation> annotations =  new ArrayList<>(asList(annotatedElement.getAnnotationsByType(singleLocator)));
```

Comment:
:) The code is very well written. Although things like this bother my internal perfectiont.